src/pages/AMM/Swap/components/SwapForm/SwapForm.tsx

```diff
@@ -38,78 +38,89 @@ const getPairChange = (pair: SwapPair, currency: CurrencyExt, name: string): Swa
   }
 };
 
+type SwapData = {
+  trade: Trade;
+  minimumAmountOut: MonetaryAmount<CurrencyExt>;
+  recipient: AddressOrPair;
+  deadline: string | number;
+};
+
+const mutateSwap = ({ deadline, minimumAmountOut, recipient, trade }: SwapData) =>
+  window.bridge.amm.swap(trade, minimumAmountOut, recipient, deadline);
+
 enum FormFields {
   INPUT_AMOUNT = 'input-amount',
   INPUT_TICKER = 'input-ticker',
   OUTPUT_AMOUNT = 'output-amount',
   OUTPUT_TICKER = 'output-ticker'
 }
 
-type SwapFormData = {
-  [FormFields.INPUT_AMOUNT]: { value: string };
-  [FormFields.INPUT_TICKER]: { value: string };
-  [FormFields.OUTPUT_AMOUNT]: { value: string };
-  [FormFields.OUTPUT_TICKER]: { value: string };
-};
-
 type Props = {
   pair: SwapPair;
+  liquidityPools: LiquidityPool[];
   onChangePair: (pair: SwapPair) => void;
+  onSwap: () => void;
 };
 
 type InheritAttrs = CardProps & Props;
 
 type SwapFormProps = Props & InheritAttrs;
 
-const SwapForm = ({ pair, onChangePair, ...props }: SwapFormProps): JSX.Element | null => {
+const SwapForm = ({ pair, liquidityPools, onChangePair, onSwap, ...props }: SwapFormProps): JSX.Element | null => {
+  const accountId = useAccountId();
   const { t } = useTranslation();
-
   const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
   const { getCurrencyFromTicker } = useGetCurrencies(bridgeLoaded);
 
-  const [slippage, setSlippage] = useState<SwapSlippage>('0.1%');
-
+  const [slippage, setSlippage] = useState(0.1);
   const [inputAmount, setInputAmount] = useState<number>();
-  const [outputAmount, setOutputAmount] = useState<number>();
+  const [trade, setTrade] = useState<Trade | null>();
```

Comment:
And here too—preferable for instances of `useState` to come first. (This is a convention/code readability thing—it feels more natural to have state right at the top, as it makes it immediately clear what values the component is going to be using and setting. Ditto with hooks following afterwards—these are the values we're calculating (`useState`; these are the values we're using to calculate state and render the component (hooks)).